tests/unit/sagemaker/serve/builder/test_js_builder.py

```diff
@@ -75,7 +75,7 @@
     "-inference:2.1.1-tgi1.4.0-gpu-py310-cu121-ubuntu20.04"
 )
 mock_djl_image_uri = (
-    "123456789712.dkr.ecr.us-west-2.amazonaws.com/djl-inference:0.24.0-neuronx-sdk2.14.1"
+    "763104351884.dkr.ecr.us-west-2.amazonaws.com/djl-inference:0.31.0-lmi13.0.0-cu124"
```

Comment:
We should add a test to ensure our original intent is satisfied - 

if pysdk_model.init_kwargs has an image more recent than the hardcoded default, then that should be the image used for optimization.